### PR TITLE
fix: update custom chart data and type only

### LINKED ISSUE #11962: [Feature]: Make custom fusion chart not to refresh on change of data
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

If I create a custom fusion chart and bind it to a widget or appsmithStore, the chart refreshes every time when the widget's value or appsmithStore value changes.
There should be some way to make this change smoother without refreshing the entire chart.

### Why should this be worked on?

chart
Custom chart config:
```
{
  "type": "cylinder",
  "dataSource": {
    "chart": {
        "subcaptionFontBold": "0",
                "cylFillColor":{{appsmith.store.power>20?"03B365":"DD4B34"}},
        "lowerLimit": "0",
        "upperLimit": "100",
        "lowerLimitDisplay": "Empty",
        "upperLimitDisplay": "Full",
        "numberSuffix": " %",
        "showhovereffect": "1",
        "theme": "fusion"
    },
    "value": {{appsmith.store.power}}
  }
}
```
This was requested by a [user in Discord](https://discord.com/channels/725602949748752515/954176816719994890).

This will be really useful in plotting some real-time data.

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Please include a summary of the changes and which issue has been fixed. Please also include relevant motivation
> and context. List any dependencies that are required for this change.

> custom fusion chart is refreshed on change of data to fixed this issue we remove chart config on `componentDidUpdate`

Fixes #11962 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/issue-11962 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 56.35 **(0)** | 37.8 **(0.01)** | 35.96 **(0)** | 56.58 **(0.01)**
 :green_circle: | app/client/src/selectors/commentsSelectors.ts | 85.25 **(1.64)** | 64.71 **(2.95)** | 73.33 **(0)** | 90.59 **(2.35)**
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0.78)** | 72.55 **(1.96)** | 100 **(0)** | 93.33 **(0.95)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :green_circle: | app/client/src/widgets/ChartWidget/component/index.tsx | 50.9 **(0.31)** | 23.23 **(-0.48)** | 54.84 **(1.71)** | 49.35 **(1.25)**</details>

### CODE REVIEW

File: app/client/src/widgets/ChartWidget/component/index.tsx [hunk 1]
```diff
@@ -437,31 +437,16 @@ class ChartComponent extends React.Component<ChartComponentProps> {
...
       this.chartInstance.chartType(chartType);
-      if (this.props.allowScroll && this.props.chartType !== "PIE_CHART") {
+      if (this.props.chartType === "CUSTOM_FUSION_CHART") {
+        const { dataSource, type } = this.getCustomFusionChartDataSource();
+        this.chartInstance.chartType(type);
+        this.chartInstance.setChartData(dataSource);
+      } else if (
+        this.props.allowScroll &&
+        this.props.chartType !== "PIE_CHART"
+      ) {
```
Comment (Tooluloope) [inline app/client/src/widgets/ChartWidget/component/index.tsx:449]: Could you add a cypress test for this?

**Review by Tooluloope — CHANGES_REQUESTED** [review]

**Comment by techbhavin** [comment]
ok-to-test sha=783d399

**Comment by YogeshJayaseelan** [comment]
@techbhavin Tested this PR and it is working as expected now. The chart doesn't refresh on choosing different data values on the fusion chart

https://www.loom.com/share/7c5e80c78c0b404ba3e04b88182db780

File: app/client/src/widgets/ChartWidget/component/index.tsx [hunk 2]
```diff
@@ -316,7 +317,7 @@ class ChartComponent extends React.Component<ChartComponentProps> {
         },
       };
     }
-    return config;
+    return config || {};
```
Comment (jsartisan) [inline app/client/src/widgets/ChartWidget/component/index.tsx:320]: Don't we need the following config like we were doing previously? Ignore if that is unecessary
```
renderAt: this.chartContainerId,
          width: "100%",
          height: "100%",
          events: {
            dataPlotClick: (evt: any) => {
              const data = evt.data;
              const seriesTitle = this.getSeriesTitle(data);
              this.props.onDataPointClick({
                x: data.categoryLabel,
                y: data.dataValue,
                seriesTitle,
              });
            },
          },
          ```

Comment (techbhavin) [reply app/client/src/widgets/ChartWidget/component/index.tsx:320]: @jsartisan 
`this.chartInstance` has those settings already so we don't have to declare it again 
in the previous case, we are declaring and re-rendering custom charts every time
to fix that scenario in the current pr, we are updating data only.

**Review by jsartisan — APPROVED** [review]

**Pull request merged**